manifests/init.pp

```diff
@@ -36,82 +32,94 @@
-    mode   => '0755',
-    source => 'puppet:///modules/puppet_metrics_collector/pe_metrics.rb'
-  }
+    file { "${scripts_dir}/puma_metrics" :
+      ensure => file,
+      mode   => '0755',
+      source => 'puppet:///modules/puppet_metrics_collector/puma_metrics'
+    }
 
-  file { "${scripts_dir}/puma_metrics" :
-    ensure => file,
-    mode   => '0755',
-    source => 'puppet:///modules/puppet_metrics_collector/puma_metrics'
-  }
+    file { "${scripts_dir}/tk_metrics" :
+      ensure => file,
+      mode   => '0755',
+      source => 'puppet:///modules/puppet_metrics_collector/tk_metrics'
+    }
 
-  file { "${scripts_dir}/tk_metrics" :
-    ensure => file,
-    mode   => '0755',
-    source => 'puppet:///modules/puppet_metrics_collector/tk_metrics'
-  }
+    exec { 'puppet_metrics_collector_daemon_reload':
+      command     => 'systemctl daemon-reload',
+      path        => ['/bin', '/usr/bin'],
+      refreshonly => true,
+    }
 
-  include puppet_metrics_collector::service::puppetserver
-  include puppet_metrics_collector::service::puppetdb
-  include puppet_metrics_collector::service::orchestrator
-  include puppet_metrics_collector::service::ace
-  include puppet_metrics_collector::service::bolt
-  include puppet_metrics_collector::service::activemq
+    include puppet_metrics_collector::service::puppetserver
+    include puppet_metrics_collector::service::puppetdb
+    include puppet_metrics_collector::service::orchestrator
+    include puppet_metrics_collector::service::ace
+    include puppet_metrics_collector::service::bolt
 
-  # LEGACY CLEANUP
+    # LEGACY CLEANUP
 
-  # Clean up old metrics directories created by the module before it was renamed.
+    # Clean up old metrics directories created by the module before it was renamed.
 
-  $legacy_dir      = '/opt/puppetlabs/pe_metric_curl_cron_jobs'
-  $safe_output_dir = shellquote($output_dir)
+    $legacy_dir      = '/opt/puppetlabs/pe_metric_curl_cron_jobs'
+    $safe_output_dir = shellquote($output_dir)
 
-  exec { "migrate ${legacy_dir} directory":
-    path    => '/bin:/usr/bin',
-    command => "mv ${legacy_dir} ${safe_output_dir}",
-    onlyif  => "[ ! -e ${safe_output_dir} -a -e ${legacy_dir} ]",
-    before  => File[$output_dir],
-  }
+    exec { "migrate ${legacy_dir} directory":
+      path    => '/bin:/usr/bin',
+      command => "mv ${legacy_dir} ${safe_output_dir}",
+      onlyif  => "[ ! -e ${safe_output_dir} -a -e ${legacy_dir} ]",
+      before  => File[$output_dir],
+    }
 
-  $legacy_files = [
-    '/opt/puppetlabs/bin/puppet-metrics-collector',
-    '/opt/puppetlabs/puppet-metrics-collector/bin',
-  ]
+    $legacy_files = [
+      '/opt/puppetlabs/bin/puppet-metrics-collector',
+      '/opt/puppetlabs/puppet-metrics-collector/bin',
+    ]
 
-  file { $legacy_files :
-    ensure => absent,
-    force  => true,
+    file { $legacy_files :
+      ensure => absent,
+      force  => true,
+    }
+  } else {
+    notify { 'systemd_provider_warning':
```

Comment:
This can cause issues with idempotency in the catalog. What was the reasoning for using a `notify` instead of a `warn` or `info` function?

Reply:
No reason other than I copied the example from the `vmware` class.